handlers/product-move-api/src/main/scala/com/gu/productmove/zuora/Subscribe.scala

```diff
@@ -0,0 +1,88 @@
+import sttp.client3.ziojson.*
+import sttp.model.Uri
+import zio.json.*
+import zio.{Clock, IO, RIO, Task, UIO, URLayer, ZIO, ZLayer}
+
+import java.time.LocalDate
+
+trait Subscribe:
+  def create(zuoraAccountId: String, targetProductId: String): ZIO[Any, String, CreateSubscriptionResponse]
+
+object SubscribeLive :
+  val layer: URLayer[ZuoraGet, Subscribe] = ZLayer.fromFunction(SubscribeLive(_))
+
+private class SubscribeLive(zuoraGet: ZuoraGet) extends Subscribe:
+  override def create(zuoraAccountId: String, targetProductId: String): ZIO[Any, String, CreateSubscriptionResponse] = {
+    for {
+      subscribeRequest <- SubscribeRequest(zuoraAccountId, targetProductId)
+      response <- zuoraGet.post[SubscribeRequest, CreateSubscriptionResponse](uri"subscriptions", subscribeRequest)
+    } yield response
+  }
+
+object Subscribe {
+  def create(zuoraAccountId: String, targetProductId: String): ZIO[Subscribe, String, CreateSubscriptionResponse] =
+    ZIO.serviceWithZIO[Subscribe](_.create(zuoraAccountId, targetProductId))
+}
+
+case class ProductRatePlanId(value: String) extends AnyVal
+case class ProductRatePlanChargeId(value: String) extends AnyVal
+
+case class CreateSubscriptionResponse(subscriptionId: String)
+given JsonDecoder[CreateSubscriptionResponse] = DeriveJsonDecoder.gen[CreateSubscriptionResponse]
+
+case class ChargeOverrides(
+                            price: Option[Double],
+                            productRatePlanChargeId: String,
+                            triggerDate: Option[LocalDate],
+                            triggerEvent: Option[String]
+                          )
+given JsonEncoder[ChargeOverrides] = DeriveJsonEncoder.gen[ChargeOverrides]
+
+case class ZuoraAccountId(value: String) extends AnyVal
+case class CaseId(value: String) extends AnyVal
+case class AcquisitionSource(value: String) extends AnyVal
+
+case class SubscribeToRatePlans(productRatePlanId: String, chargeOverrides: List[ChargeOverrides] = List())
+given JsonEncoder[SubscribeToRatePlans] = DeriveJsonEncoder.gen[SubscribeToRatePlans]
+
+case class SubscribeRequest(
+                             accountKey: String,
+                             autoRenew: Boolean = true,
+                             contractEffectiveDate: LocalDate,
+                             customerAcceptanceDate: LocalDate,
+                             termType: String = "TERMED",
+                             renewalTerm: Int = 12,
+                             initialTerm: Int = 12,
+                             subscribeToRatePlans: List[SubscribeToRatePlans],
+                             AcquisitionCase__c: String,
+                             AcquisitionSource__c: String,
+                             CreatedByCSR__c: String
+                           )
+object SubscribeRequest {
+  given JsonEncoder[SubscribeRequest] = DeriveJsonEncoder.gen[SubscribeRequest]
+
+  def apply(zuoraAccountId: String, targetProductId: String): ZIO[Any, Nothing, SubscribeRequest] =
```

Comment:
I'd actually be tempted not to use `apply` here as it's not sucha simple constructor.  How about withTodaysDate?
```suggestion
  def withTodaysDate(zuoraAccountId: String, targetProductId: String): ZIO[Any, Nothing, SubscribeRequest] =
```
Another option to explore would be to pass in the dates as a normal parameter which would make this normal scala rather than ZIO.

Reply:
I made it a ZIO so I could add an easy unit test for it in `SubscribeSpec.scala`, otherwise if we added it at a higher-level in the `create` function where it's being called, we'd have to mock the zuoraGet layer I think.

Reply:
we can discuss if you like, I don't have strong feelings if it's simpler to keep it as a ZIO.  I do have stronger feelings on the name, nowadays I think apply should generally only be used for fairly trivial constructors, but we can also discuss that if you like.

Reply:
Yep, I agree on the naming, have updated it.